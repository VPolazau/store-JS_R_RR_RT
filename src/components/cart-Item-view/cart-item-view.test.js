import React from 'react'
import renderer from 'react-test-renderer'

import CartItemView from './cart-item-view'

test('test CartItemView', () => {
  const mockInfo = {
    img: 'https://i.dummyjson.com/data/products/1/1.jpg',
    title: 'iPhone 9',
    count: 1,
    price: 549,
  }

  const component = renderer.create(
    <CartItemView
      info={mockInfo}
      isDeleted={false}
      inc={() => {}}
      dec={() => {}}
      removeItem={() => {}}
    />
  )

  let componentTree = component.toJSON()
  expect(componentTree).toMatchSnapshot()
})
