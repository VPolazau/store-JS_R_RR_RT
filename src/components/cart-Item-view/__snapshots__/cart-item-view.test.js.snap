// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CartItemView 1`] = `
<div
  className="CartItemView"
>
  <div
    className="imageUrl"
  >
    <img
      alt="cartItemViewImage"
      className="image"
      src="https://i.dummyjson.com/data/products/1/1.jpg"
    />
  </div>
  <div
    className="info"
  >
    <div
      className="title"
    >
      iPhone 9
    </div>
    <div
      className="info-btns"
    >
      <div
        className="counter"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium counter-btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RemoveOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </button>
        1
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium counter-btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </button>
      </div>
      <div
        className="result-price"
      >
        Payable price: 
        Â 
        549
        $
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium btn-del-cart css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        Delete
      </button>
    </div>
  </div>
</div>
`;
